jmx_prometheus_httpserver/src/main/java/io/prometheus/jmx/WebServer.java

```diff
@@ -14,14 +16,24 @@ public static void main(String[] args) throws Exception {
 
      String[] hostnamePort = args[0].split(":");
      int port;
-
+     InetSocketAddress socket;
+     
      if (hostnamePort.length == 2) {
        port = Integer.parseInt(hostnamePort[1]);
+       socket = new InetSocketAddress(hostnamePort[0], port);
      } else {
        port = Integer.parseInt(hostnamePort[0]);
+       socket = new InetSocketAddress(port);
      }
 
-     JmxCollector jc = new JmxCollector(new File(args[1])).register();
-     HTTPServer server = new HTTPServer(port);
+     new JmxCollector(new File(args[1])).register();
+     final HTTPServer server = new HTTPServer(socket, CollectorRegistry.defaultRegistry);
+     
+     Runtime.getRuntime().addShutdownHook(new Thread() {
```

Comment:
We don't need this, the httpserver should keep going until the process dies.

There could be a long time between the TERM and the process actually dying.